solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -1867,35 +1873,79 @@ public ScoreMode scoreMode() {
       qr.setNextCursorMark(cmd.getCursorMark());
       hitsRelation = Relation.EQUAL_TO;
     } else {
-      final TopDocsCollector<?> topCollector = buildTopDocsCollector(len, cmd);
-      MaxScoreCollector maxScoreCollector = null;
-      Collector collector = topCollector;
-      if ((cmd.getFlags() & GET_SCORES) != 0) {
-        maxScoreCollector = new MaxScoreCollector();
-        collector = MultiCollector.wrap(topCollector, maxScoreCollector);
+      if (log.isInfoEnabled()) {
+        log.info("calling from 2, query: {}", query.getClass());
       }
-      ScoreMode scoreModeUsed =
-          buildAndRunCollectorChain(qr, query, collector, cmd, pf.postFilter).scoreMode();
+      MTCollectorQueryCheck allowMT = new MTCollectorQueryCheck();
```

Comment:
Bunch of code being added in two places looking similar.  Have you thought of attempting to factor out a common approach?

Reply:
https://github.com/apache/solr/pull/2248/commits/c69006a4ca6625307e17fb0c272b08eebacd9a69 and https://github.com/apache/solr/pull/2248/commits/1ec21d4274fb10eaeec2fed04093b91cf1e88c72 factored something out. Not thought much about naming in doing so and not specifically in response to this comment but just from my code reading/learning/reviewing perspective.